app/src/main/java/org/mozilla/fenix/components/accounts/FenixAccountManager.kt

```diff
@@ -21,6 +21,20 @@ open class FenixAccountManager(context: Context) {
     val accountProfileEmail
```

Comment:
`authenticatedAccount` seem to have valid uses, like [here](https://github.com/mozilla-mobile/fenix/blob/bf3f81d271382eff0780cfdd81f6dea35c13dc7f/app/src/main/java/org/mozilla/fenix/components/toolbar/DefaultToolbarMenu.kt#L336)

`signedToFxa` was used in a similar buggy way for the home menu. Switched that to use `accountState`,  the same used in browser menu.
Since that was it's only use I've removed this method.

Reply:
Maybe we should also refactor `accountProfileEmail` to encapsulate the  `authenticatedAccount` usage to really discourage it from being used erroneously.  For the `accountProfileEmail` getter, I imagine we can call the `authenticatedAccount` check and just return a nullable string for the email.

That way we can simplify `getSyncItemTitle` to 
```
val accountManager.accountProfileEmail()

return if (!email.isNullOrEmpty()) {            
   email        
} else {            
   context.getString(R.string.sync_menu_sign_in)        
}
```

We probably don't want to ever return the email if the account is not authenticated. I think this will allow us remove the `authenticatedAccount` getter completely or make it private to really discourage it.

```
fun getAccountProfileEmail() : String? {
  return if (accountManager.authenticatedAccount() != null) {
    accountManager.accountProfile()?.email
 } else {
   null
}
```

Reply:
Refactored `accountProfileEmail` to only try and return the email for the account if that is in the authenticated state.
Having this check here, in one place, means consumers will not need to double check the account status.
This also allowed removing the `authenticatedAccount` property from `FenixAccountManager` which now has only two methods, more as helpers for interacting with an account: `accountProfileEmail` and `accountState`.